Jenkinsfile

```diff
@@ -0,0 +1,147 @@
+/*
+ * (C) Copyright 2021 Nuxeo (http://nuxeo.com/).
+ * This is unpublished proprietary source code of Nuxeo SA. All rights reserved.
+ * Notice of copyright on this source code does not indicate publication.
+ *
+ * Contributors:
+ *     Julien Carsique <jcarsique@nuxeo.com>
+ */
+
+@Library('nxAILibUntrusted@feat-INSIGHT-988-pipelineLib') _
+
+pipeline {
+    agent {
+        label "jenkins-ai-nuxeo1010"
+    }
+    options {
+        disableConcurrentBuilds()
+        buildDiscarder(logRotator(daysToKeepStr: '60', numToKeepStr: '60', artifactNumToKeepStr: '5'))
+        timeout(time: 1, unit: 'HOURS')
+    }
+    environment {
+        VERSION = ''
+        MARKETPLACE_URL = 'https://connect.nuxeo.com/nuxeo/site/marketplace'
+        MARKETPLACE_URL_PREPROD = 'https://nos-preprod-connect.nuxeocloud.com/nuxeo/site/marketplace'
+        PACKAGE_PATTERN = 'nuxeo-jira-ingestion-package/target/nuxeo-jira-ingestion-package-*.zip'
+    }
+    stages {
+        stage('Init') {
+            steps {
+                container('nuxeo1010') {
+                    script {
+                        stepsInit()
+                    }
+                }
+            }
+            post {
+                always {
+                    setGitHubBuildStatus('init')
+                }
+            }
+        }
+        stage('Maven Build') {
+            steps {
+                container('nuxeo1010') {
```

Comment:
wondering how soon we'll need it to be 11.x/2021.x and if it needs to be a var

Reply:
Yes, the agent label and container name are almost the only difference with 11.x/2021.
It would be a var if I could factorize the stage, steps or container parts but that seems very hard (if not impossible) in declarative mode. I was limiting the factorization to still being able to customize the behavior and use "replay" being able to live edit the pipeline code.
I can move the whole pipeline in the library. Let's give a try.